_episodes/31.md

```diff
@@ -0,0 +1,579 @@
+              name: config-volume
+            - mountPath: /etc/trino/catalog
+              name: catalog-volume
+          ports:
+            - name: http
+              containerPort: 8080
+              protocol: TCP
+          livenessProbe:
+            httpGet:
+              path: /v1/info
+              port: http
+          readinessProbe:
+            httpGet:
+              path: /v1/info
+              port: http
+          resources:
+            {}
+```
+
+### ConfigMap
+
+You may have noticed that the pods define volumes that are referring to an
+object called [ConfigMap](https://kubernetes.io/docs/concepts/configuration/configmap/).
+This is a way to store non-confidential data in the form of key-value pairs.
+
+This is how the Trino chart loads the [Trino configurations](https://trino.io/docs/current/installation/deployment.html#configuring-trino) 
+in the `/etc/trino` directory on the containers. This ConfigMap file, 
+`trino/templates/configmap-worker.yaml`, defines the files loaded into the 
+worker nodes. The only real difference between the worker ConfigMap is in the 
+`config.properites` file specifying if the node is a coordinator or not. 
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: tcb-trino-worker
+  labels:
+    app: trino
+    chart: trino-0.3.0
+    release: tcb
+    heritage: Helm
+    component: worker
+data:
+  node.properties: |
+    node.environment=production
+    node.data-dir=/data/trino
+    plugin.dir=/usr/lib/trino/plugin
+
+  jvm.config: |
+    -server
+    -Xmx8G
+    -XX:+UseG1GC
+    -XX:G1HeapRegionSize=32M
+    -XX:+UseGCOverheadLimit
+    -XX:+ExplicitGCInvokesConcurrent
+    -XX:+HeapDumpOnOutOfMemoryError
+    -XX:+ExitOnOutOfMemoryError
+    -Djdk.attach.allowAttachSelf=true
+    -XX:-UseBiasedLocking
+    -XX:ReservedCodeCacheSize=512M
+    -XX:PerMethodRecompilationCutoff=10000
+    -XX:PerBytecodeRecompilationCutoff=10000
+    -Djdk.nio.maxCachedBufferSize=2000000
+
+  config.properties: |
+    coordinator=false
+    http-server.http.port=8080
+    query.max-memory=4GB
+    query.max-memory-per-node=1GB
+    query.max-total-memory-per-node=2GB
+    memory.heap-headroom-per-node=1GB
+    discovery.uri=http://tcb-trino:8080
+
+  log.properties: |
+    io.trino=INFO
+```
+
+The only other ConfigMap actually defines the [catalog files](https://trino.io/docs/current/installation/deployment.html#catalog-properties)
+in the `/etc/trino/catalog` folder. This ConfigMap only contains the two TPC
+benchmark connectors added.
+
+```
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: tcb-trino-catalog
+  labels:
+    app: trino
+    chart: trino-0.3.0
+    release: tcb
+    heritage: Helm
+    role: catalogs
+data:
+  tpch.properties: |
+    connector.name=tpch
+    tpch.splits-per-node=4
+  tpcds.properties: |
+    connector.name=tpcds
+    tpcds.splits-per-node=4
+```
+
+### Networking
+
+Unlike in the Docker world, where a Docker container always has an ip address, 
```

Comment:
IP address .. here and everywhere in this post

Reply:
This sentence really is wrong .. The difference is that docker runs on the host directly and you can expose the container. In k8s the pod runs in the cluster and networking is private..